### PR TITLE
Merge v1.9.0-beta03

### PR DESCRIPTION
**Pull request by str4d**
This merges current `main` (specifically 150778f0083008dfde5fe464c4f44d39014f7d84) into the ZIP 316 feature branch.

# Author

- [ ] Self-review: Did you review your own code in GitHub's web interface? _Code often looks different when reviewing the diff in a browser, making it easier to spot potential bugs._
- [ ] Automated tests: Did you add appropriate automated tests for any code changes?
- [ ] Manual tests: Did you update the [manual tests](../blob/main/docs/tests) as appropriate? _While we aim for automated testing of the application, some aspects require manual testing. If you had to manually test something during development of this pull request, write those steps down._
- [ ] Documentation: Did you update documentation as appropriate? (e.g [README.md](../blob/main/README.md), etc.)
- [ ] Run the app: Did you run the demo app and try the changes?
- [ ] Rebase and squash: Did you pull in the latest changes from the main branch and squash your commits before assigning a reviewer? _Having your code up to date and squashed will make it easier for others to review. Use best judgement when squashing commits, as some changes (such as refactoring) might be easier to review as a separate commit._

# Reviewer

- [x] Checklist review: Did you go through the code with the [Code Review Guidelines](../blob/main/docs/CODE_REVIEW_GUIDELINES.md) checklist?
- [x] Ad hoc review: Did you perform an ad hoc review?  _In addition to a first pass using the code review guidelines, do a second pass using your best judgement and experience which may identify additional questions or comments. Research shows that code review is most effective when done in multiple passes, where reviewers look for different things through each pass._
- [ ] Automated tests: Did you review the automated tests?
- [ ] Manual tests: Did you review the manual tests?
- [ ] How is Code Coverage affected by this PR? _We encourage you to compare coverage before and after changes and when possible, leave it in a better place._
- [ ] Documentation: Did you review Docs, [README.md](../blob/main/README.md), and [Architecture.md](../blob/main/docs/Architecture.md) as appropriate?
- [ ] Run the app: Did you run the demo app and try the changes? _While the CI server runs the app to look for build failures or crashes, humans running the app are more likely to notice unexpected log messages, UI inconsistencies, or bad output data. Perform this step last, after verifying the code changes are safe to run locally._

### CODE REVIEW

**Review by nuttycom — APPROVED** [review]
utACK

The only question I had was with respect to the transparent address derivation methods in sdk-lib/src/main/java/cash/z/ecc/android/sdk/tool/DerivationTool.kt - I presume that these will be removed in favor of unified-key-oriented methods?

**Comment by str4d** [comment]
> The only question I had was with respect to the transparent address derivation methods in sdk-lib/src/main/java/cash/z/ecc/android/sdk/tool/DerivationTool.kt - I presume that these will be removed in favor of unified-key-oriented methods?

Not necessarily; the SDK may still need to deal in legacy transparent addresses for certain interoperability. #530 didn't remove those APIs because they weren't broken by the migration to ZIP 316. In any case, this PR is irrelevant to this particular point as it is just resolving merge conflicts.

**Review by ccjernigan — APPROVED** [review]
This looks fine.

We do have one more large PR that was just merged to the main branch today, so we might want to integrate that into the feature branch ASAP as well.

File: README.md [hunk 1]
```diff
@@ -16,212 +16,42 @@ This is a beta build and is currently under active development. Please be advise
...
-There are two special checkpoints, one for sapling activation and another for orchard activation.  These are mentioned because they don't follow the "round 10,000" rule.
- * Sapling activation
-     * Mainnet: 419200
-     * Testnet: 280000
- * Orchard activation
-     * Mainnet: 1687104
-     * Testnet: 1842420
-
-## Publishing
+There are also more comprehensive sample walletes:
```
Comment (pacu) [inline README.md:39]:
```suggestion
There are also more comprehensive sample wallets:
```

**Review by pacu — COMMENTED** [review]
[Non-Blocking] typo on Readme

File: README.md (hunk 1 above)
Comment (ccjernigan) [reply README.md:39]: It's French for `wallets` 😂.

This is a pre-existing issue in the main branch.

Comment (str4d) [reply README.md:39]: That's a typo in the `main` branch; it should be fixed there.

**Comment by str4d** [comment]
The failing test `test_android_modules_wtf` is unrelated to this PR; it is failing tests in the feature branch due to test vectors that need updating (to use UFKVs etc). I'm working on a separate PR for that.

**Pull request merged**